docs/master/eloquent/complex-where-conditions.md

```diff
@@ -129,10 +155,18 @@ input QueryPeopleWhereWhereConditionsRelation {
   "Additional condition logic."
   condition: QueryPeopleWhereWhereConditions
```

Comment:
I realized that this is nonsensical when combined with `columnsEnum`. The allowed columns of the original model are used for the relations as well.

We need to find a solution that either has a separate input that allows columns as strings for relations, or an even more sophisticated solution that has enums for the relations. Configuration could look like this:

```graphql
@whereConditions(relations: [
  { name: "friends", columns: ["name", "age"] }
  { name: "friends.friends", columns: ["name", "age"] }
])

input QueryPeopleWhereWhereConditionsRelation {
  friends: QueryPeopleWhereWhereConditionsFriendsConditions
  friendsFriends: QueryPeopleWhereWhereConditionsFriendsFriendsConditions
}

input QueryPeopleWhereWhereConditionsFriendsConditions{
  column: QueryPeopleWhereWhereConditionsFriendsColumn
  operator: SQLOperator = EQ
  value: Mixed
  AND: [QueryPeopleWhereWhereConditionsFriendsConditions!]
  OR: [QueryPeopleWhereWhereConditionsFriendsConditions!]
}

enum QueryPeopleWhereWhereConditionsFriendsConditions {
  NAME
  AGE
}
```